test/saithrift/tests/saitunnel.py

```diff
@@ -0,0 +1,341 @@
+        #  create router interface,
+        #  create neighbor
+        #  create nhop
+        #  create route
+        #   
+        ############################################################################## 
+		
+        #ingress router interface
+        ingress_if_id=sai_thrift_create_router_interface(self.client,vr_id=vr_id,is_port=1,port_id=port1,vlan_id=vlan_id,v4_enabled = 1,v6_enabled = 1,is_lb=0,mac='')
+        #ingress neighbor (ip=10.10.1.1 , mac=00:05:06:00:00:00 )
+        sai_thrift_create_neighbor(self.client,addr_family=addr_family,rif_id=ingress_if_id,ip_addr=ingress_ip_addr,dmac=neighbor_mac_ingress)
+        #ingress(encap) nhop and route create
+        ingress_nhop=sai_thrift_create_nhop(self.client,addr_family=addr_family,is_tunnel=0,ip_addr=ingress_nhop_ip_addr,rif_id=ingress_if_id)
+        sai_thrift_create_route(self.client,vr_id=vr_id,addr_family=addr_family,ip_addr=nhop_ip_addr_ingress,ip_mask=nhop_ip_mask_ingress,nhop=ingress_nhop)
+	   
+        ###############################################################################
+        #  Ingress configurations
+        #  create router interface,
+        #  create neighbor
+        #  create nhop
+        #  create route
+        #   
+        ############################################################################## 
+        	
+        #egress router interface
+        egress_if_id=sai_thrift_create_router_interface(self.client,vr_id=vr_id,is_port=1,port_id=port2,vlan_id=vlan_id,v4_enabled=1,v6_enabled = 1,is_lb=0,mac='')
+        #egress neighbor (ip=20.20.1.1 , mac = 00:05:06:00:00:02)
+        sai_thrift_create_neighbor(self.client,addr_family=addr_family,rif_id=egress_if_id,ip_addr=egress_ip_addr,dmac=neighbor_mac_egress)
+        #egress nhop and route create
+        egress_nhop=sai_thrift_create_nhop(self.client,addr_family=addr_family,is_tunnel=0,ip_addr=egress_nhop_ip_addr,rif_id=egress_if_id)
+        sai_thrift_create_route(self.client,vr_id=vr_id,addr_family=addr_family,ip_addr=nhop_ip_addr_egress,ip_mask=nhop_ip_mask_egress,nhop=egress_nhop)
+        #adding tunnel and route
+        initiator_id=sai_thrift_create_nhop(self.client,addr_family=addr_family,is_tunnel=1,ip_addr=initiator_ip_addr,rif_id=tunnel_id)
+        sai_thrift_create_route(self.client,vr_id=vr_id,addr_family=addr_family,ip_addr=tunnel_ip_addr,ip_mask=tunnel_ip_mask,nhop=initiator_id)
+        
+        
+		
+        ###################################################################################
+        #Creating a Tunnel Table entry with
+        #Source match 10.10.1.10
+        #Dest match 2.2.2.2
+        #
+        #####################################################################################
+        
+        #create tunnel table entry for incoming match 		
+        tunnel_entry_id=sai_thrift_create_tunnel_table(self.client,addr_family=addr_family,vr_id=vr_id,ip_addr_dst=ip_addr_decap_dst,ip_addr_src=ip_addr_decap_src,tunnel_oid=tunnel_id)
+        

```

Comment:
great test cases! Though I think it would be cool if we can also test for when we have multiple tunnel table entries related to one tunnel_id and making sure that all packets with the dst ips are decapped. Also, for decap, it might be helpful to test to make sure that we decap packets from multiple test src ips and making sure the packet is decapped as well as long as it matches the dst ip.